x-pack/test/api_integration/apis/siem/feature_controls.ts

```diff
@@ -0,0 +1,297 @@
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import expect from '@kbn/expect';
+import gql from 'graphql-tag';
+import { SecurityService, SpacesService } from '../../../common/services';
+import { KbnTestProvider } from './types';
+
+const introspectionQuery = gql`
+  query Schema {
+    __schema {
+      queryType {
+        name
+      }
+    }
+  }
+`;
+
+const featureControlsTests: KbnTestProvider = ({ getService }) => {
+  const supertest = getService('supertestWithoutAuth');
+  const security: SecurityService = getService('security');
+  const spaces: SpacesService = getService('spaces');
+  const clientFactory = getService('siemGraphQLClientFactory');
+
+  const expectGraphQL404 = (result: any) => {
+    expect(result.response).to.be(undefined);
+    expect(result.error).not.to.be(undefined);
+    expect(result.error).to.have.property('networkError');
+    expect(result.error.networkError).to.have.property('statusCode', 404);
+  };
+
+  const expectGraphQLResponse = (result: any) => {
+    expect(result.error).to.be(undefined);
+    expect(result.response).to.have.property('data');
+    expect(result.response.data).to.be.an('object');
+  };
+
+  const expectGraphIQL404 = (result: any) => {
+    expect(result.error).to.be(undefined);
+    expect(result.response).not.to.be(undefined);
+    expect(result.response).to.have.property('statusCode', 404);
+  };
+
+  const expectGraphIQLResponse = (result: any) => {
+    expect(result.error).to.be(undefined);
+    expect(result.response).not.to.be(undefined);
+    expect(result.response).to.have.property('statusCode', 200);
+  };
+
+  const executeGraphQLQuery = async (username: string, password: string, spaceId?: string) => {
+    const queryOptions = {
+      query: introspectionQuery,
+    };
+
+    const basePath = spaceId ? `/s/${spaceId}` : '';
+
+    const client = clientFactory({ username, password, basePath });
+    let error;
+    let response;
+    try {
+      response = await client.query(queryOptions);
+    } catch (err) {
+      error = err;
+    }
+    return {
+      error,
+      response,
+    };
+  };
+
+  const executeGraphIQLRequest = async (username: string, password: string, spaceId?: string) => {
+    const basePath = spaceId ? `/s/${spaceId}` : '';
+
+    return supertest
+      .get(`${basePath}/api/siem/graphql/graphiql`)
+      .auth(username, password)
+      .then((response: any) => ({ error: undefined, response }))
+      .catch((error: any) => ({ error, response: undefined }));
+  };
+
+  describe('feature controls', () => {
+    it(`APIs can't be accessed by user with logstash-* "read" privileges`, async () => {
+      const username = 'logstash_read';
+      const roleName = 'logstash_read';
+      const password = `${username}-password`;
+      try {
+        await security.role.create(roleName, {
+          elasticsearch: {
```

Comment:
We added `elasticsearch` privileges for the Infra api integration tests, because all of the GraphQL queries didn't succeed without them. The API calls which you all using doesn't require these, so we can remove the `elasticsearch` section for all of the roles created in this file.